### DISCUSSION TITLE
Hello, I really like Navidrome and I want to make some simple modifications to generate a Docker image. How can I achieve this?

### DISCUSSION THREAD

**Discussion by digua-cn** [navidrome/navidrome#2599] [Q&A]
Hello, I really like Navidrome and I want to make some simple modifications to generate a Docker image. How can I achieve this?

**Comment by digua-cn** [comment]
In other words, I have now cloned the Navidrome project, made some simple modifications, and would like to build a Docker image. How can I go about building it?

**Reply by deluan** [reply]
I just created a new make target to build docker images, in https://github.com/navidrome/navidrome/commit/7f551a79328e0781b78d537aa88abcb1d4e9442a

You just have to run `make docker` from the command line. It builds a `linux/amd64` image (tagged as `deluan/navidrome:develop`)

**Comment by deluan** [comment]
You'd need to create a `Dockerfile` to build the image. You may use our old (optimized and complex) `Dockerfile` as a starting point: https://github.com/navidrome/navidrome/blob/50f3a2c11df90b69a62c36bf41c62a23275a84d4/Dockerfile

Another option from @kgarner7 from [Discord](https://discord.com/channels/671335427726114836/704303730660737113/1163973581894594650):

> FYI, if you want to make production builds of your image, you can actually do that using Github.
> You'd fork deluan/navidrome, and then you'd need to configure three repository secrets:
> - DOCKER_IMAGE: (YOUR_USERNAME/navidrome)
> - DOCKER_USERNAME, DOCKER_PASSWORD: (I'd personally recommend using an access token)
> 
> With that, you can push to master and it'll eventually deploy to github and docker
